sample: fs_sample: Build is failed for the sample.filesystem.fat_fs test case for the mm_feather board.

**Describe the bug**

The sample.filesystem.fat_fs test case defined in samples\subsys\fs\fs_sample\sample.yaml must be supported by all boards with "sdhc" :
```
  sample.filesystem.fat_fs:
    depends_on: sdhc
```
But `west build -p -b mm_feather samples/subsys/fs/fs_sample -T sample.filesystem.fat_fs` is failed

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b mm_feather samples/subsys/fs/fs_sample -T sample.filesystem.fat_fs
2.  See error:

```
-- Configuring done (32.6s)
-- Generating done (5.0s)
-- Build files have been written to: C:/_ddm/ZEPHYR/zephyr_main/zephyr/build
-- west build: building application
[1/152] Generating include/generated/zephyr/version.h
-- Zephyr version: 4.1.99 (C:/_ddm/ZEPHYR/zephyr_main/zephyr), build: v4.1.0-3141-g8b3eb368c67b
[104/152] Building C object modules/fatfs/CMakeFiles/modul...s.dir/C_/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c.obj
FAILED: modules/fatfs/CMakeFiles/modules__fatfs.dir/C_/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c.obj
C:\Users\nxa12205\zephyr-sdk-0.17.0\arm-zephyr-eabi\bin\arm-zephyr-eabi-gcc.exe -DBOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024 -DCPU_MIMXRT1062DVL6A -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DXIP_BOOT_HEADER_DCD_ENABLE=1 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -DZEPHYR_CONFIG_OVERRIDE=zephyr_fatfs_config.h -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/build/zephyr/include/generated/zephyr -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/include -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/build/zephyr/include/generated -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/soc/nxp/imxrt -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/soc/nxp/imxrt/imxrt10xx/. -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/soc/nxp/imxrt/. -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/soc/nxp/imxrt/imxrt10xx -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/cmsis/CMSIS/Core/Include -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/cmsis/. -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/common -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/igpio -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/lpuart -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/ocotp -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/cache/armv7-m7 -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/devices/RT/RT1060/MIMXRT1062 -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/devices/RT/RT1060/periph -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/devices/RT/RT1050/MIMXRT1052/xip -IC:/_ddm/ZEPHYR/zephyr_main/modules/hal/nxp/mcux/mcux-sdk-ng/devices/RT/RT1060/MIMXRT1062/drivers -IC:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs -IC:/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/include -isystem C:/_ddm/ZEPHYR/zephyr_main/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros C:/_ddm/ZEPHYR/zephyr_main/zephyr/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=C:/Users/nxa12205/zephyr-sdk-0.17.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros C:/_ddm/ZEPHYR/zephyr_main/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=C:/_ddm/ZEPHYR/zephyr_main/zephyr/samples/subsys/fs/fs_sample=CMAKE_SOURCE_DIR -fmacro-prefix-map=C:/_ddm/ZEPHYR/zephyr_main/zephyr=ZEPHYR_BASE -fmacro-prefix-map=C:/_ddm/ZEPHYR/zephyr_main=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT modules/fatfs/CMakeFiles/modules__fatfs.dir/C_/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c.obj -MF modules\fatfs\CMakeFiles\modules__fatfs.dir\C_\_ddm\ZEPHYR\zephyr_main\modules\fs\fatfs\ff.c.obj.d -o modules/fatfs/CMakeFiles/modules__fatfs.dir/C_/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c.obj -c C:/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c
C:/_ddm/ZEPHYR/zephyr_main/modules/fs/fatfs/ff.c:463:2: error: #error Wrong FF_VOLUMES setting
  463 | #error Wrong FF_VOLUMES setting
      |  ^~~~~
[114/152] Building C object modules/fatfs/CMakeFiles/modules__fatfs.dir/zfs_diskio.c.obj
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c: In function 'disk_status':
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:31:46: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   31 |         if (disk_access_status(PDRV_STR_ARRAY[pdrv]) != 0) {
      |                                              ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c: In function 'disk_read':
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:51:44: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   51 |         if (disk_access_read(PDRV_STR_ARRAY[pdrv], buff, sector, count) != 0) {
      |                                            ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c: In function 'disk_write':
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:63:45: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   63 |         if (disk_access_write(PDRV_STR_ARRAY[pdrv], buff, sector, count) != 0) {
      |                                             ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c: In function 'disk_ioctl':
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:125:61: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
  125 |                         if (disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_CTRL_INIT, NULL) !=
      |                                                             ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:119:61: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
  119 |                         if (disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_CTRL_DEINIT, NULL) !=
      |                                                             ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:107:53: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
  107 |                 if (disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_GET_ERASE_BLOCK_SZ, buff) !=
      |                                                     ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:97:54: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   97 |                 if ((disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_GET_SECTOR_SIZE,
      |                                                      ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:86:53: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   86 |                 if (disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_GET_SECTOR_COUNT, buff) !=
      |                                                     ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:80:53: warning: array subscript <unknown> is outside array bounds of 'const char * const[0]' [-Warray-bounds]
   80 |                 if (disk_access_ioctl(PDRV_STR_ARRAY[pdrv], DISK_IOCTL_CTRL_SYNC, buff) != 0) {
      |                                                     ^
C:/_ddm/ZEPHYR/zephyr_main/zephyr/modules/fatfs/zfs_diskio.c:22:26: note: while referencing 'pdrv_str'
   22 | static const char *const pdrv_str[] = {FF_VOLUME_STRS};
      |                          ^~~~~~~~
[125/152] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Users\nxa12205\.mcuxpressotools\cmake-3.28.1-windows-x86_64\bin\cmake.EXE' --build 'C:\_ddm\ZEPHYR\zephyr_main\zephyr\build'
```

**Expected behavior**
Successful build.

**Impact**
CI is failed.

**Environment (please complete the following information):**

 - Commit SHA: 8b3eb368c67b0523573d75c0116d5b8c86481ef2

--- a/boards/madmachine/mm_feather/mm_feather.dts
+++ b/boards/madmachine/mm_feather/mm_feather.dts
@@ -171,6 +171,12 @@
 	pinctrl-2 = <&pinmux_usdhc1_med>;
 	pinctrl-3 = <&pinmux_usdhc1_fast>;
 	pinctrl-names = "default", "slow", "med", "fast";
+
+	mmc {
+		compatible = "zephyr,sdmmc-disk";
+		disk-name = "SD";
+		status = "okay";
+	};
 };
 
 &edma0 {

--- a/modules/fatfs/zephyr_fatfs_config.h
+++ b/modules/fatfs/zephyr_fatfs_config.h
@@ -113,6 +113,7 @@
 	DT_FOREACH_STATUS_OKAY(zephyr_flash_disk, _FF_DISK_NAME) \
 	DT_FOREACH_STATUS_OKAY(zephyr_ram_disk, _FF_DISK_NAME) \
 	DT_FOREACH_STATUS_OKAY(zephyr_sdmmc_disk, _FF_DISK_NAME) \
+	DT_FOREACH_STATUS_OKAY(zephyr_mmc_disk, _FF_DISK_NAME) \
 	DT_FOREACH_STATUS_OKAY(st_stm32_sdmmc, _FF_DISK_NAME)
 
 #undef FF_VOLUMES

